src/main/java/mate/academy/spring/validation/EmailValidator.java

```diff
@@ -0,0 +1,22 @@
+package mate.academy.spring.validation;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class EmailValidator implements ConstraintValidator<Email, String> {
+    private static final String EMAIL_VALIDATOR
+            = "^([a-zA-Z0-9_\\-\\.]+)@((\\[[0-9]{1,3}\\.[0-9]{1,3}\\"
+            + ".[0-9]{1,3}\\.)|(([a-zA-Z0-9\\-]+\\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\\]?)$";
+
+    @Override
+    public boolean isValid(String email, ConstraintValidatorContext constraintValidatorContext) {
+        if (email == null) {
+            return false;
+        }
+        Pattern pattern = Pattern.compile(EMAIL_VALIDATOR);
+        Matcher matcher = pattern.matcher(email);
+        return matcher.matches();
```

Comment:
can be just this instead of 2 more objects
```suggestion
        return email.matches(EMAIL_VALIDATOR);
```